code/modules/economy/holopay.dm

```diff
@@ -0,0 +1,249 @@
+/// Max range at which the hologram can be projected before it deletes
+#define MAX_HOLO_RANGE 4
+/// Minimum forced fee for holopay stations. Registers as "pay what you want."
+#define MIN_FEE 0
+/// Maximum forced fee. It's unlikely for a user to encounter this type of money, much less pay it willingly.
+#define MAX_FEE 5000
+
+/obj/machinery/holopay
+	name = "holographic pay stand"
+	desc = "an unregistered pay stand"
+	icon = 'icons/obj/economy.dmi'
+	icon_state = "card_scanner"
+	alpha = 150
+	anchored = TRUE
+	layer = FLY_LAYER
+	/// Owner of the holopay
+	var/mob/living/owner
+	/// ID linked to the holopay
+	var/obj/item/card/id/linked_card
+	/// Owner's linked account to the holopay
+	var/datum/bank_account/linked_account
+	/// Replaces the "pay whatever" functionality with a set amount when non-zero.
+	var/force_fee = 0
+	/// List of logos available for customization - via font awesome 5
+	var/static/list/available_logos = list("angry", "ankh", "band-aid", "cannabis", "cat", "cocktail", "coins", "comments-dollar",
+	"cross", "cut", "donate", "dna", "flask", "glass-cheers", "glass-martini-alt", "hand-holding-usd", "heart", "heart-broken",
+	"hamburger", "hat-cowboy-side", "money-check-alt", "music", "pizza-slice", "prescription-bottle-alt", "radiation", "robot", "smile",
+	"tram", "trash")
+	/// The brand icon chosen by the user
+	var/shop_logo = "donate"
+
+/obj/machinery/holopay/New(turf/location, mob/living/user, obj/item/card/id/card, datum/bank_account/account)
+	loc = location
+	owner = user
+	linked_card = card
+	linked_account = account
+	desc = "Pays directly into [owner]'s bank account."
+	add_atom_colour("#77abff", FIXED_COLOUR_PRIORITY)
+	set_light(2)
+	visible_message(span_notice("A holographic pay stand appears."))
+	/// Start checking if the owner has left or died
+	RegisterSignal(owner, list(COMSIG_MOVABLE_MOVED, COMSIG_LIVING_STATUS_UNCONSCIOUS), .proc/check_operation)
+	return ..()
+
+/**
+ * A periodic check to see if the owner has left the area, died, or is afk.
+ * Deletes the holopay if any condition is true.
+ */
+/obj/machinery/holopay/proc/check_operation()
+	SIGNAL_HANDLER
+	if(!owner || !owner.client || !isliving(owner) || !IN_GIVEN_RANGE(src, owner, MAX_HOLO_RANGE))
+		QDEL_NULL(linked_card?.my_store)
+		qdel(src)
+
+/obj/machinery/holopay/Destroy()
+	playsound(loc, "sound/effects/empulse.ogg", 40, TRUE)
+	visible_message(span_notice("The pay stand vanishes."))
+	return ..()
```

Comment:
Destroy is meant to be used for cleanup, not effects. You should do this in a proc you call when you want to make effects that also qdels the holopay machine

Reply:
Kinda on that topic, you need to clear the linked card's holopay machine reference here. that or make it a weakref